rust/main/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -315,6 +315,7 @@ where
         &self,
         message: &HyperlaneMessage,
         metadata: &[u8],
+        apply_gas_overrides: bool,
```

Comment:
nit: maybe we can have a `ContractCallOptions` struct that holds different parameters. I think this makes the code a bit easier to read compared to a boolean function parameter.
But I also see some places that don't have `tx_gas_estimate`, so not sure.

```rust
#[derive(Clone, Debug)]
pub struct ContractCallOptions {
    pub apply_gas_overrides: bool,
    pub tx_gas_estimate: Option<U256>,
}

async fn process_contract_call(
    &self,
    message: &HyperlaneMessage,
    metadata: &[u8],
    options: &ContractCallOptions,
) -> ChainResult<ContractCall<M, ()>> {
    ...
}
```